lib/hooks/core/hook-http.js

```diff
@@ -51,13 +51,13 @@ function requestWrap(request) {
 
     var namespace = cls.getNamespace();
     var uri = uriFromOptions(options);
-    var currContext = cls.getRootContext().createChildSpanData(uri);
-    currContext.span.setLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
-    currContext.span.setLabel(TraceLabels.HTTP_URL_LABEL_KEY, uri);
+    var span = agent.startSpan(uri);
+    span.addLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
```

Comment:
I definitely can pass them to `agent.startSpan`. Do you want `addLabel` to be removed as well? I don't have a strong preference but if you don't want it used then it should be removed.